langkit/textstat.py

```diff
@@ -1,131 +1,220 @@
 from logging import getLogger
+from typing import Callable, List, Tuple
 import textstat
-from whylogs.core.datatypes import String
-from whylogs.experimental.core.metrics.udf_metric import register_metric_udf
+from whylogs.experimental.core.udf_schema import register_dataset_udf
+from . import LangKitConfig
+
+lang_config = LangKitConfig()
+
+
+_udfs_to_register: List[Tuple[Callable, str]] = []
+
+
+def langkit_udf(schema_name="") -> Callable:
+    def decorator_register(func):
+        global _udfs_to_register
+        _udfs_to_register.append((func, schema_name))
+        return func
+
+    return decorator_register
 
 
 diagnostic_logger = getLogger(__name__)
 
 # score metrics
 
 
-@register_metric_udf(col_type=String, schema_name="text_standard_component")
-def flesch_kincaid_grade(text: str) -> float:
-    return textstat.textstat.flesch_kincaid_grade(text)
+@langkit_udf("text_standard_component")
+def flesch_kincaid_grade(text):
+    result = []
+    for input in text:
```

Comment:
```suggestion
    index = (
        text.columns[0] if isinstance(text, pd.DataFrame) else list(text.keys())[0]
    )
    for input in text[index]:
```
This needs to be fixed in all the UDFs. It is worrying that the unit tests didn't catch this

Reply:
I should just make a wrapper function so we can do
```
@langkit_udf(textstat.textstat.flesch_kincaid_grade, "optional_schema_name")
def flesch_kincaid_grade(text):
    return wrapper(f, text)
```

Reply:
Or maybe just a list of `(function, schema_name)` in `init()`...